### PR TITLE
EG-1182 - UI components and creation logic for the Details component

### PR DESCRIPTION
**Pull request by LesWilson**

### CODE REVIEW

File: app/services/UIBuilder.scala [hunk 1]
```diff
@@ -18,11 +18,9 @@ package services
...
 import javax.inject.Singleton
 import models._
-import models.ocelot.stanzas.{CurrencyPoundsOnlyInput, DateInput, Question, Input, CurrencyInput, _}
-import models.ocelot.{Phrase, Link}
-import models.ui.{stackStanzas, UIComponent, Page, Text, Table, Paragraph}
-import models.ui.{CyaSummaryList, NameValueSummaryList, Answer, ErrorMsg, H1, H2, H3, H4, InsetText}
-import models.ui.{ConfirmationPanel, TypeErrorMsg, RequiredErrorMsg, ValueErrorMsg, BulletPointList}
+import models.ocelot.stanzas.{CurrencyInput, CurrencyPoundsOnlyInput, DateInput, Input, Question, _}
+import models.ocelot.{Link, Phrase}
+import models.ui.{Answer, BulletPointList, ConfirmationPanel, CyaSummaryList, Details, ErrorMsg, H1, H2, H3, H4, InsetText, NameValueSummaryList, Page, Paragraph, RequiredErrorMsg, Table, Text, TypeErrorMsg, UIComponent, ValueErrorMsg, stackStanzas}
```
Comment (cmrhmrc) [inline app/services/UIBuilder.scala:23]: greater than max 160 char line length

File: app/services/UIBuilder.scala [hunk 2]
```diff
@@ -189,4 +191,11 @@ class UIBuilder {
     val texts: Seq[Text] = ycg.group.map(c => TextBuilder.fromPhrase(c.text))
     ConfirmationPanel(texts.head, texts.tail)
   }
+  private def fromSectionAndNoteGroup(caption: Text, ng: NoteGroup)(implicit stanzaIdToUrlMap: Map[String, String]): UIComponent = {
```
Comment (cmrhmrc) [inline app/services/UIBuilder.scala:194]: can probably remove the implicit stanzaIdToUrlMap off these and moany other from*() fns

Comment (cmrhmrc) [inline app/services/UIBuilder.scala:194]: Dont need the extra braces on these two fns

File: app/services/UIBuilder.scala (hunk 1 above)
Comment (LesWilson) [reply app/services/UIBuilder.scala:23]: Will be updated in next push

File: app/services/UIBuilder.scala (hunk 2 above)
Comment (LesWilson) [reply app/services/UIBuilder.scala:194]: as discussed, still needed due to TextBuilder

Comment (LesWilson) [reply app/services/UIBuilder.scala:194]: removed

**Pull request merged**